### PR TITLE
Debugging level logging list inspection

### PR DESCRIPTION
**Pull request by burnout87**

### CODE REVIEW

File: oda_api/api.py [hunk 1]
```diff
@@ -816,7 +816,7 @@ def get_product(self,
             instrument=instrument, product_type=product), cookies=self.cookies)
 
         if res.status_code != 200:
-            warnings.warn(
```
Comment (volodymyrss) [inline oda_api/api.py:819]: although we did not use `warnings` much, it's not the same as `logging.warning`. See [here](https://docs.python.org/3/howto/logging.html). 

```
`warnings.warn()` in library code if the issue is avoidable and the client application should be modified to eliminate the warning

`logging.warning()` if there is nothing the client application can do about the situation, but the event should still be noted
```

In this case, we actually want the client to do something: adapt parameters. So, following the python convention, it should be `warnings.warn`

File: oda_api/api.py [hunk 2]
```diff
@@ -841,9 +841,9 @@ def get_product(self,
                         msg += '%s' % valid_names
                         msg += '\n'
                         msg += 'this will throw an error in a future version \n'
-                        msg += 'and might breack the current request!\n '
```
Comment (volodymyrss) [inline oda_api/api.py:844]: I think it's `break` here.

File: oda_api/api.py [hunk 3]
```diff
@@ -841,9 +841,9 @@ def get_product(self,
...
+                        msg += 'and might brake the current request!\n '
                         msg += '----------------------------------------------------------------------------\n'
-                        warnings.warn(msg)
```
Comment (volodymyrss) [inline oda_api/api.py:846]: this is actually a warning. And, following note above, not `logging.warn` bit `warnings.warn`

**Review by volodymyrss — CHANGES_REQUESTED** [review]

File: oda_api/api.py (hunk 3 above)
Comment (burnout87) [reply oda_api/api.py:846]: According to what we discussed in #59 , I thought this was supposed to be changed to debug level message. Or perhaps I misunderstood.

Comment (volodymyrss) [reply oda_api/api.py:846]: > According to what we discussed in #59 , I thought this was supposed to be changed to a debug level message. Or perhaps I misunderstood.

The part that lead to `unable to dig list, instance not a dict by...` should be.  This is utterly incomprehensible for the user.

`invalid parameter` warning should remain. You can imagine it is important that user is warned (at least!) when using incorrect parameter name.
It should be separately determined why, as you noted as well, `T_format` is not understood. It should be, in principle, understood and used correctly. Maybe not in all queries though.
I guess it was passed by frontend API.
In any case, at this level, it should be clear to user that this parameter is not used.

Comment (burnout87) [reply oda_api/api.py:846]: Ok, now is clear. I was confused and thought we were referring to the warning message that should have been changed.

As of `T_format`, I had indeed the same impression.

**Review by volodymyrss — APPROVED** [review]
thanks! please rename though.

**Pull request merged**